File: content/src/main/java/org/sandboxpowered/silica/minecraft/blocks/NoteBlock.java
package org.sandboxpowered.silica.minecraft.blocks;

import org.sandboxpowered.api.block.BaseBlock;
import org.sandboxpowered.api.block.Block;
import org.sandboxpowered.api.state.BlockState;
import org.sandboxpowered.api.state.Properties;
import org.sandboxpowered.api.state.StateFactory;
import org.sandboxpowered.api.state.property.EnumProperty;

public class NoteBlock extends BaseBlock {
    private static final EnumProperty<Instrument> INSTRUMENT = EnumProperty.of("instrument", Instrument.class);

    public NoteBlock(Settings settings) {
        super(settings);
    }

    @Override
    public void appendProperties(StateFactory.Builder<Block, BlockState> builder) {
        super.appendProperties(builder);
        builder.add(INSTRUMENT, Properties.NOTE, Properties.POWERED);
    }

    public static enum Instrument {
        HARP,
        BASEDRUM,
        SNARE,
        HAT,
        BASS,
        FLUTE,
        BELL,
        GUITAR,
        CHIME,
        XYLOPHONE,
        IRON_XYLOPHONE,
        COW_BELL,
        DIDGERIDOO,
        BIT,
        BANJO,
        PLING
    }
}

File: common/src/main/kotlin/org/sandboxpowered/silica/block/NoteBlock.kt
package org.sandboxpowered.silica.block

import org.sandboxpowered.silica.state.StateProvider
import org.sandboxpowered.silica.state.block.BlockState
import org.sandboxpowered.silica.util.Identifier

class NoteBlock(identifier: Identifier) : BaseBlock(identifier) {
    override fun appendProperties(builder: StateProvider.Builder<Block, BlockState>) {
        super.appendProperties(builder)
        builder.add(SilicaBlockProperties.NOTE, SilicaBlockProperties.POWERED, SilicaBlockProperties.INSTRUMENT)
    }
}